File: Lv.2/기능개발.cs
//기능개발
//2022.08.14.
//흠...나중에 다시 풀어보자...
int[] solution(int[] progresses, int[] speeds)
{
    int day = 0;
    int max_day = 0;

    var result = new List<int>();
    for (int i = 0; i < progresses.Length; ++i)
    {
        day = GetDay(progresses[i], speeds[i]);
        if (max_day < day)
        {
            result.Add(1);
        }
        else
        {
            result[^1]++;
        }

        if (max_day < day)
        {
            max_day = day;
        }
    }

    return result.ToArray();
}

int GetDay(int progress, int speed)
{
    return (int)Math.Ceiling((100f - progress) / speed);
}

File: 프로그래머스/코딩테스트 고득점 Kit/스택,큐/기능개발.cpp
#include <bits/stdc++.h>
using namespace std;
vector<int> solution(vector<int> progresses, vector<int> speeds)
{
    vector<int> answer;
    for (int i = 0; i < progresses.size();)
    {
        
        int day = ceil((100 - progresses[i]) / (double)speeds[i]);

        for (int j = i; j < progresses.size(); j++)
        {
            progresses[j] += speeds[j] * day;
        }

        int temp = 0;
        do{
            temp++;
            i++;
        } while (i < progresses.size() && !(progresses[i] < 100));

        answer.push_back(temp);
    }

    return answer;
}


//와... day 처리나, ++answer.back();허허헣..
/*
vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;

    int day;
    int max_day = 0;
    for (int i = 0; i < progresses.size(); ++i)
    {
        day = (99 - progresses[i]) / speeds[i] + 1;

        if (answer.empty() || max_day < day)
            answer.push_back(1);
        else
            ++answer.back();

        if (max_day < day)
            max_day = day;
    }

    return answer;
}
*/